### PR TITLE
Adds recipe for mccortex

### PR DESCRIPTION
**Pull request by winni2k**
* [X] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [X] This PR adds a new recipe.
* [X] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

**Comment by winni2k** [comment]
@bioconda/core: I think this PR is ready to merge.

File: recipes/mccortex/build.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+set -x
+
+export CFLAGS="-I$PREFIX/include"
+export LDFLAGS="-L$PREFIX/lib"
+export CPATH=${PREFIX}/include
+
+for make_file in libs/string_buffer/Makefile $(find libs/seq_file -name Makefile) $(find libs/seq-align -name Makefile) Makefile; do
+    sed -i.bak 's/-lz/-lz $(LDFLAGS)/' $make_file
+done
+
```
[the remark this answers is not in the document]
Comment (winni2k) [reply recipes/mccortex/build.sh]: Mccortex has extensive dependencies that are all part of the source code release.  The patching is applied to only a small number of all the make files in the mccortex source code. I need autoreconf for other make files...

[the remark this answers is not in the document]
Comment (winni2k) [reply recipes/mccortex/build.sh]: Happy to fix the other issues in an upstream PR.

File: recipes/mccortex/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+source:
+  url: https://github.com/mcveanlab/mccortex/releases/download/mccortex-{{ version }}/mccortex{{ version_uscore }}.tar.gz
+  sha256: '79a74b5e20646a7d7a4e5c7927bca58beeded97df9889f13fe33c126331cf886'
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+  host:
+    - autoconf
+    - automake
```
[the remark this answers is not in the document]
Comment (winni2k) [reply recipes/mccortex/meta.yaml]: Will do, but see comment below.

File: recipes/mccortex/build.sh [hunk 3]
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
+set -x
+
+export CFLAGS="-I$PREFIX/include"
+export LDFLAGS="-L$PREFIX/lib"
+export CPATH=${PREFIX}/include
```
[the remark this answers is not in the document]
Comment (winni2k) [reply recipes/mccortex/build.sh]: Sure! But someone should probably fix the docs where I copied this from: https://bioconda.github.io/troubleshooting.html#zlib-errors

[the remark this answers is not in the document]
Comment (epruesse) [reply recipes/mccortex/build.sh]: Agreed. See bioconda/bioconda.github.io#22

**Comment by epruesse** [comment]
Thanks @winni2k

**Pull request merged**